### PR TITLE
make it possible to use external Aeron media driver, #20588

### PR DESCRIPTION
**Pull request by patriknw**
- on my machine the MaxThroughputSpec maxed out all 8 cores completely,
  and when using external media driver it is much better and easier to
  find the actual bottlenecks

You can start the external media driver with:

```
sbt "akka-remote/runMain io.aeron.driver.MediaDriver /Users/patrik/tmp/aeron-throughput.properties"
```

where the properties file may look like:

```
aeron.mtu.length=16384
aeron.socket.so_sndbuf=2097152
aeron.socket.so_rcvbuf=2097152
aeron.rcv.buffer.length=16384
aeron.rcv.initial.window.length=2097152
agrona.disable.bounds.checks=true

aeron.dir=/Users/patrik/tmp/aeron-data
aeron.threading.mode=SHARED_NETWORK
```

Note that by using shared threading mode it seems to consume less cpu and I did not notice any difference in performance of the `MaxThroughputSpec`.

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #20588

**Comment by drewhk** [comment]
Why not store the properties in `test/resources`? That way we can share the same and it is not mixed with released code.

File: akka-remote/src/main/resources/reference.conf [hunk 1]
```diff
@@ -105,6 +105,18 @@ akka {
       # stream, to pass such messages through the large message stream the selections
       # but must be resolved to ActorRefs first.
       large-message-destinations = []
+      
+      advanced {
+        # Start the Aeron media driver in the same JVM. Set to 'off' when using external
```
Comment (drewhk) [inline akka-remote/src/main/resources/reference.conf]: "Controls whether to start the ..."

**Comment by drewhk** [comment]
LGTM. Can we make the Jenkins perf-job using this? Although that seems a bit scary since a separate process needs to be managed with a build-run.

**Comment by patriknw** [comment]
yes, I can put the aeron.properties in test/resources

I don't think external media driver will improve things on jenkins perf-job because there we use two machines (multi-node), instead of running sender and receiver on same

**Comment by drewhk** [comment]
ok

**Comment by ktoso** [comment]
LGTM

**Comment by patriknw** [comment]
added the [aeron.properties](https://github.com/akka/akka/pull/20653/commits/dc80ddef661c3c879971b432d961558302a57a93)

It can now be started with:
```
sbt "akka-remote/test:runMain io.aeron.driver.MediaDriver aeron.properties"
```

**Comment by ktoso** [comment]
LGTM, merge at will 👍

**Pull request merged**